debug_toolbar/middleware.py

```diff
@@ -32,6 +39,17 @@ def get_show_toolbar():
         return func_or_path
 
 
+@lru_cache()
+def get_observe_request():
```

Comment:
Since this function isn't called in middleware, it shouldn't live here. I think moving it to the utils module or adding it to DebugToolbar would be more apt. The same with observe_request.